chord_metadata_service/restapi/authz_middleware.py

```diff
@@ -48,6 +59,65 @@ def __call__(self, request):
 
                 allowed_datasets = response.json()["result"]
                 request.allowed_datasets = allowed_datasets
+        # auth for rego_dev_playground repo
+        elif (CANDIG_OPA_VERSION == "rego_dev_playground" and
+              any(re.match(path_re, request.path) for path_re in AUTHZ_PATHS) and
+              request.method == "GET"):
+            tokens = {}
+            for header in request.headers:
+                header_all_caps = header.upper()
+                if header_all_caps.startswith(("X-CANDIG-LOCAL-", "X-CANDIG-DAC-", "X-CANDIG-FED-", "X-CANDIG-EXT-")):
+                    tokens[header] = json.loads(request.headers[header])
```

Comment:
I noticed that this code block is duplicated in two places:
```
            tokens = {}
            for header in request.headers:
                header_all_caps = header.upper()
                if header_all_caps.startswith(("X-CANDIG-LOCAL-", "X-CANDIG-DAC-", "X-CANDIG-FED-", "X-CANDIG-EXT-")):
                    tokens[header] = json.loads(request.headers[header])
```
To make the code more readable, I think creating a helper function for this code and then calling the function whenever it's necessary would make the code neater.